spark/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -166,12 +178,21 @@ case class DeltaParquetFileFormat(
       }
       results.headOption.map(e => ColumnMetadata(e._2, e._1))
     }
+
     val isRowDeletedColumn = findColumn(IS_ROW_DELETED_COLUMN_NAME)
-    val rowIndexColumn = findColumn(ROW_INDEX_COLUMN_NAME)
+    val rowIndexColumnName = if (predicatePushdownEnabled) {
+      ParquetFileFormat.ROW_INDEX_TEMPORARY_COLUMN_NAME
+    } else {
+      ROW_INDEX_COLUMN_NAME
+    }
+    val rowIndexColumn = findColumn(rowIndexColumnName)
 
-    if (isRowDeletedColumn.isEmpty && rowIndexColumn.isEmpty) {
-      return parquetDataReader // no additional metadata is needed.
+    // No additional metadata is needed.
```

Comment:
This is getting complicated. 

Can `rowIndexColumn` be non-empty when `predicatePushdown` is true?

This check can be just its own
```
if (!rowIndexColumn.isEmpty) {
      // verify the file splitting and filter pushdown are disabled. The new additional
      // metadata columns cannot be generated with file splitting and filter pushdowns
      require(!isSplittable, "Cannot generate row index related metadata with file splitting")
      require(disablePushDowns, "Cannot generate row index related metadata with filter pushdown")
}
```